### ISSUE TITLE
[Bug] OnPageLoad execution order is incorrect

### ISSUE THREAD

**Issue by Nikhil-Nandagopal** [appsmithorg/appsmith#2562]
## Description

A single query fails intermittently on page load. This query fetch_monthly_dead_apps depends on another query and should run once it's execution is complete
https://app.appsmith.com/applications/5ffeeadc5b452f525458d4d5/pages/5ffeeadc5b452f525458d4e1/edit

### Steps to reproduce the behaviour:
1. Refresh the page multiple times till you see an error in the 4th table for dead apps

The layout action response is incorrect in this case. There are 2 problems
1. fetch_monthly_active_apps is repeated twice in this call stack
2. all the parallel actions are in the second array position which will delay all other action calls. Instead only the dead apps should be in the second position

```[
  [
    {
      "id": "5ffeeadd5b452f525458d4e9",
      "name": "fetch_monthly_active_apps",
      "pluginType": "DB",
      "jsonPathKeys": [
        "moment().add(-1,'months').format(\"YYYY-MM\")"
      ],
      "timeoutInMillisecond": 10000
    }
  ],
  [
    {
      "id": "5ffeeadd5b452f525458d4ed",
      "name": "fetch_monthly_dead_apps",
      "pluginType": "DB",
      "jsonPathKeys": [
        "moment().add(-2,'months').format(\"YYYY-MM\")",
        "fetch_monthly_active_apps.data.map((app) => \"'\" + app.active_app + \"'\").join(\",\")"
      ],
      "timeoutInMillisecond": 10000
    },
    {
      "id": "5ffeeadd5b452f525458d4e9",
      "name": "fetch_monthly_active_apps",
      "pluginType": "DB",
      "jsonPathKeys": [
        "moment().add(-1,'months').format(\"YYYY-MM\")"
      ],
      "timeoutInMillisecond": 10000
    },
    {
      "id": "5ffeeadd5b452f525458d50f",
      "name": "fetch_weekly_active_apps",
      "pluginType": "DB",
      "jsonPathKeys": [
        "moment().add(-30,'days').format(\"YYYY-WW\")"
      ],
      "timeoutInMillisecond": 10000
    }
  ]
]```

**Comment by mohanarpit** [comment]
Duplicate of #2524. Closing the other issue because this has more details
